addons/hr/populate/hr.py

```diff
@@ -78,20 +82,50 @@ class HrEmployee(models.Model):
     _inherit = 'hr.employee'
 
     _populate_sizes = {'small': 100, 'medium': 2000, 'large': 20000}
-    _populate_dependencies = ['hr.department', 'hr.job', 'hr.work.location', 'hr.employee.category']
+    _populate_dependencies = [
+        'res.company',  # multi-company setup
+        'res.users',
+        'resource.calendar',
+        'hr.department', 'hr.job', 'hr.work.location', 'hr.employee.category']
 
     def _populate(self, size):
         employees = super()._populate(size)
         self._populate_set_manager(employees)
+        self._populate_set_users_companies(employees)
         return employees
 
     def _populate_factories(self):
-        company_id = self.env.ref('base.main_company').id
+        company_ids = self.env.registry.populated_models['res.company']
+
+        company_calendars = {}
+        for company_id in company_ids:
+            company_calendars[company_id] = (
+                self.env["resource.calendar"]
+                .search([('company_id', '=', company_id),
+                         ('name', 'not like', 'Standard')])).ids
+
         department_ids = self.env.registry.populated_models['hr.department']
         job_ids = self.env.registry.populated_models['hr.job']
         work_location_ids = self.env.registry.populated_models['hr.work.location']
         tag_ids = self.env.registry.populated_models['hr.employee.category']
 
+        admin_id = self.env.ref('base.user_admin').id
+        user_ids = filter(
+            lambda uid: uid != admin_id,
+            self.env.registry.populated_models['res.users'])
+
+        def _compute_user_and_company(iterator, *args):
+            # First users then as many as required non-users
+            for values, company_id, user_id in itertools.zip_longest(
+                    iterator, itertools.cycle(company_ids), user_ids,
+                    fillvalue=False):
+                yield {'company_id': company_id,
+                       'user_id':    user_id,
+                       **values}
```

Comment:
Wouldn't it be better to use the user's company instead? :thinking: 
You also avoid updating each user that way.